app/src/components/widgets/charts/launchExecutionAndIssueStatistics/issueStatisticsChart.jsx

```diff
@@ -0,0 +1,250 @@
+ * along with Report Portal.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import 'c3/c3.css';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { injectIntl, intlShape } from 'react-intl';
+import * as d3 from 'd3-selection';
+import classNames from 'classnames/bind';
+import * as COLORS from 'common/constants/colors';
+import ReactDOMServer from 'react-dom/server';
+import { TooltipWrapper } from '../common/tooltip';
+import { C3Chart } from '../common/c3chart';
+import chartStyles from './launchExecutionAndIssueStatistics.scss';
+import tooltipStyles from './launchExecutionAndIssueStatisticsTooltip.scss';
+import { Legend } from './launchExecutionAndIssuesChartLegend';
+import { LaunchExecutionAndIssueStatisticsTooltip } from './launchExecutionAndIssueStatisticsTooltip';
+import { getPercentage, getDefectItems, messages } from './chartUtils';
+
+const chartContext = classNames.bind(chartStyles);
+const tooltipContext = classNames.bind(tooltipStyles);
+
+@injectIntl
+export class IssueStatisticsChart extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    widget: PropTypes.object.isRequired,
+    isPreview: PropTypes.bool,
+    container: PropTypes.instanceOf(Element).isRequired,
+    observer: PropTypes.object.isRequired,
+  };
+
+  static defaultProps = {
+    isPreview: false,
+    height: 0,
+  };
+
+  state = {
+    isConfigReady: false,
+  };
+
+  componentDidMount() {
+    this.props.observer.subscribe('widgetResized', this.resizeIssuesChart);
+    this.getConfig();
+  }
+  componentWillUnmount() {
+    this.issuesNode.removeEventListener('mousemove', this.setCoords);
+    this.props.observer.unsubscribe('widgetResized', this.resizeIssuesChart);
+  }
+
+  onIssuesChartCreated = (chart, element) => {
+    this.issuesChart = chart;
+    this.issuesNode = element;
+    if (!this.props.widget.content.result || this.props.isPreview) {
+      return;
+    }
+
+    d3
+      .select(chart.element)
+      .select('.c3-chart-arcs-title')
+      .attr('dy', -15)
+      .append('tspan')
+      .attr('dy', 30)
+      .attr('x', 0)
+      .attr('fill', '#666')
+      .text('ISSUES');
+
+    this.issuesNode.addEventListener('mousemove', this.setCoords);
+  };
+
+  onIssuesMouseOut = () => {
+    this.issuesChart.revert();
+  };
+
+  onIssuesMouseOver = (id) => {
+    this.issuesChart.focus(id);
+  };
+
+  onIssuesClick = (id) => {
+    this.issuesChart.toggle(id);
+  };
+
+  getConfig = () => {
+    const { widget, container } = this.props;
+    const defectTypesChartData = {};
+    const defectTypesChartDataOrdered = [];
+    const defectTypesChartColors = {};
+    const values = widget.content.result[0].values;
+
+    this.height = container.offsetHeight;
+    this.width = container.offsetWidth;
+    this.noAvailableData = false;
+
+    Object.keys(values).forEach((key) => {
+      if (!key.includes('$executions$')) {
```

Comment:
For this case, you can find some utils in `components/widgets/charts/common`

Reply:
fixed